### PR TITLE
Improved the behavior of the delimiter the next_word method

### PR DESCRIPTION
**Pull request by reugene**
Improved the behavior of the delimiter the next_word method
next_date now have separator choice the default remains '.'
Tested the hell out of the parser. Oh yeah!

### CODE REVIEW

**Comment by reugene** [comment]
@nicpottier @ericnewcomer plz review
Just reminding myself the good workflow we use to have.

File: nsms/parser/tests.py [hunk 1]
```diff
@@ -112,3 +176,23 @@ def test_parsing(self):
...
+        self.assertEquals("reg", parser.next_keyword(["reg"]))
+        self.assertEquals("James", parser.next_word())
+        self.assertEquals("Kirk", parser.next_word())
+        self.assertEquals(datetime.date(day=10, month=12, year=1944), parser.next_date())
+        self.assertEquals("0788383381", parser.next_phone())
+
+        self.assertFalse(parser.has_word())
+
+        parser = Parser("REG. James. Kirk. 10/12/44. 0788383381", '.')
+
```
Comment (nyaruka) [inline nsms/parser/tests.py:248]: Wouldn't this be a better test as:  Parser("REG.James.Kirk.10/12/44.0788383381", '.')

File: nsms/parser/tests.py [hunk 2]
```diff
@@ -112,3 +176,23 @@ def test_parsing(self):
         self.assertEquals("0788383381", parser.next_phone())
 
         self.assertFalse(parser.has_word())
+
+        parser = Parser("REG, James, Kirk, 10.12.44, 0788383381", ',')
+
```
Comment (nyaruka) [inline nsms/parser/tests.py:238]: Same here, better test would be without the space (or a mix of both)

File: nsms/parser/tests.py [hunk 3]
```diff
@@ -34,73 +34,137 @@ def test_word_parsing(self):
...
         self.assertNextDate(23, 6, 2011, "23.06.11")
+        self.assertNextDate(23, 6, 2011, "23/06/11", separator='/')
+        self.assertNextDate(23, 6, 2011, "23-06-11", separator='-')
+
         self.assertNextDate(23, 6, 2000, "23.06.00")
+        self.assertNextDate(23, 6, 2000, "23/06/00", separator='/')
+        self.assertNextDate(23, 6, 2000, "23-06-00", separator='-')
 
         # invalid day
```
Comment (nyaruka) [inline nsms/parser/tests.py:179]: So for dates I think we should support any of the three characters as separators.. that 'word' of the date will be split based on the Parser separator, so whatever the word is we have we should support splitting it based on one of '.' '-' or '/'

Comment (reugene) [inline nsms/parser/tests.py:179]: I am not sure I am getting your point correctly, but what I did is to support these characters and any other that could come up depending on the user. That can be "*" or "|", what I mean is that you never know what people really come up with.

I can easily support only these three separators, what I did I just go beyond them and give the user the chance to be creative.:)

**Comment by norkans7** [comment]
@nicpottier @ericnewcomer please review

**Pull request merged**